### PR TITLE
Docker: pull image before building

### PR DESCRIPTION
**Pull request by jnewb1**
to prevent cache invalidation across buildjet/github

### CODE REVIEW

File: Dockerfile.openpilot_base [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:20.04@sha256:ed4a42283d9943135ed87d4ee34e542f7f5ad9ecf2f244870e23122f703f91c2
```
Comment (adeebshihadeh) [inline Dockerfile.openpilot_base:1]: what about pulling the latest? we don't want to have to manually update this

Comment (jnewb1) [reply Dockerfile.openpilot_base:1]: ubuntu:20.04 *should* pull the latest version, but github has locally cached versions that seem to override this. I'll see if I can find a better way to do this

Comment (adeebshihadeh) [reply Dockerfile.openpilot_base]: can't we `docker pull` it?

Comment (jnewb1) [reply Dockerfile.openpilot_base]: ah, yea I think we just needed to add --pull to buildx build

**Comment by jnewb1** [comment]
resolved the cache invalidation: https://github.com/commaai/openpilot/actions/runs/7108057226/job/19350988928

**Pull request merged**